Bottom: e63611f9872eb3391adc5976eb9e48b8dd06b830
Top:    8e17ce44d915a61a37514e7d8adbefa73fe03378
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-19 17:47:03 +0100

Refresh of hid-common-don-t-walk-off-end-

---

diff --git a/src/hid/common/actions.c b/src/hid/common/actions.c
index 0256772..4bd886b 100644
--- a/src/hid/common/actions.c
+++ b/src/hid/common/actions.c
@@ -358,7 +358,8 @@ another:
   while (*sp && *sp != '(')
     *cp++ = *sp++;
   *cp++ = 0;
-  sp++;
+  if (*sp)
+    sp++;
 
   /*
    * we didn't find a leading ( so invoke the action
